src/builder.ts

```diff
@@ -12,18 +12,20 @@ export function composePropertyDecorators(decorators: PropertyDecorator[]): Prop
 }
 
 /* A factory function for a builder.
- *
- * Used as input to a chain of mixins.
- *
- * TypeScript doesn't allow passing a typed class (e.g. `Builder<Options>`) to a mixin,
- * but does allow passing a typed function call that returns a class (e.g. `createBuilder<Options>()`)
  */
-export function createBuilder<Options>(): Constructor<Builder<Options>> {
-    return class BaseBuilder implements Builder<Options> {
+export function createBuilder<Options>(
+    ...initializers: BuilderInitializer<Options>[]
```

Comment:
The builder factor now takes a list of initializers rather than being wrapped with a set of mixins.